src/lwc/geGiftEntryFormApp/geGiftEntryFormApp.js

```diff
@@ -65,6 +71,37 @@ export default class GeGiftEntryFormApp extends NavigationMixin(LightningElement
         this.namespace = getNamespace(DATA_IMPORT_BATCH_OBJECT.objectApiName);
     }
 
+    connectedCallback() {
+        registerListener('geBatchGiftEntryTableChangeEvent', this.retrieveBatchTotals, this);
+    }
+
+    disconnectedCallback() {
+        unregisterListener('geBatchGiftEntryTableChangeEvent', this.retrieveBatchTotals, this);
+    }
+
+    async retrieveBatchTotals() {
+        this.batchTotals = await BatchTotals(this.batchId);
+
+        if(this.batchTotals.hasPaymentsWithExpiredAuthorizations &&
+            !this._hasPaymentsWithExpiredAuthorizations) {
+            const detail = {
```

Comment:
This can also be extracted into its own function, we could call it `buildModalDetail`

Reply:
@voduyemi Do you think it's better to build the detail in the function and then call the dispatch event here, or should the function be something like `displayExpiredAuthorizationWarningModal` and handle the dispatch of the togglemodal event as well?

Reply:
I think I like the idea of naming it `displayExpiredAuthorizationWarningModal` and handling and dispatching in that function